.github/workflows/hlint.yaml

```diff
@@ -0,0 +1,29 @@
+name: Check hlint
+
+on:
+  pull_request:
+
+jobs:
+  hlint:
+    name: Check hlint
+    runs-on: ubuntu-latest
+    steps:
+    - name: 📥 Checkout repository
+      uses: actions/checkout@v4
+
+    - name: ❄ Prepare nix
+      uses: cachix/install-nix-action@v25
+      with:
+        extra_nix_config: |
+          accept-flake-config = true
+          log-lines = 1000
+
+    - name: ❄ Cachix cache of nix derivations
+      uses: cachix/cachix-action@v14
+      with:
+        name: cardano-scaling
+        authToken: '${{ secrets.CACHIX_CARDANO_SCALING_AUTH_TOKEN }}'
+
+    - name: 📐 Check hlint
+      run: |
+        nix develop .#fmt --command hlint .
```

Comment:
Should: Just expand the formatting workflow

We use the exact same shell as in the `formatting` workflow, we should avoid preparing it twice. So I would suggest to move it as an additional step on the existing workflow. 

In fact, the code formatting job in the formatting workflow should actually be even part of the `ci-nix` workflow as we have also the exact same workflow trigger.